yt/frontends/gadget/data_structures.py

```diff
@@ -363,24 +363,20 @@ def _parse_parameter_file(self):
         self.domain_dimensions = np.ones(3, "int32")
         self._periodicity = (True, True, True)
 
-        self.cosmological_simulation = 1
+        self.current_redshift = hvals.get("Redshift", 0.0)
+        if "Redshift" not in hvals:
+            mylog.info("Redshift is not set in Header. Assuming z=0.")
 
-        try:
-            self.current_redshift = hvals["Redshift"]
-        except KeyError:
-            # Probably not a cosmological dataset, we should just set
-            # z = 0 and let the user know
-            self.current_redshift = 0.0
-            only_on_root(mylog.info, "Redshift is not set in Header. Assuming z=0.")
-
-        try:
+        if "OmegaLambda" in hvals:
             self.omega_lambda = hvals["OmegaLambda"]
             self.omega_matter = hvals["Omega0"]
             self.hubble_constant = hvals["HubbleParam"]
-        except KeyError:
+            self.cosmological_simulation = self.omega_lambda != 0.0
+        else:
             # If these are not set it is definitely not a cosmological dataset.
             self.omega_lambda = 0.0
-            self.omega_matter = 1.0  # Just in case somebody asks for it.
+            self.omega_matter = 0.0  # Just in case somebody asks for it.
```

Comment:
Why change this to 0.0, instead if 1.0? This would imply there's nothing in the volume.

Reply:
From @neutrinoceros 
> non-cosmological datasets normally have omega_matter set to 0.0. Is there a reason not to follow the convention here ?

in PR #4470

Reply:
Ok, so it was changed for consistency with other frontends? That's ok, but maybe warrants changing the comment from 'Just incase somebody asks for it' to 'Changed to 0.0 for consistency with non-cosmological frontends' in case somebody comes looking?

Reply:
Fair point. The comment was actually left unchanged from the previous code (and I believe is just referencing that omega_matter is being set, not necessarily it's value), I had only changed the numerical value here. 

I've added your comment, though I left the original as well, since it does describe why omega_matter is being set at all.